### PR TITLE
News api handler

### PR DESCRIPTION
**Pull request by KenmogneThimotee**
## Description

Please include a summary of the change and the issue it solves. 
I have implemented a news API integration
#5645 
**Fixes** #(issue)

## Type of change

(Please delete options that are not relevant)

- [x] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?

(Describe at a high level how the feature was implemented)

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).

### CODE REVIEW

File: mindsdb/integrations/handlers/newsapi_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# NewsAPI API Handler
+
+This handler integrates with the [News API](https://newsapi.org/docs) to make aggregate article (data available to use for model training and predictions.
+
+## Example: Selet artcles from news api
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/newsapi_handler/README.md:5]: Can you please fix the typo in Select

File: mindsdb/integrations/handlers/newsapi_handler/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Connect to the NewsAPI API
+
+We start by creating a database to connect to the News API.
+
+```
+CREATE DATABASE newsAPI
+WITH
+  ENGINE = 'newsAPI'
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/newsapi_handler/README.md:14]: Let's keep the engine name lowercase

File: mindsdb/integrations/handlers/newsapi_handler/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+FROM newsAPI.article
+WHERE query = 'mindsdb'
+AND sources="abc-news"
+AND publishedAt >= "2023-03-23" AND  publishedAt <= "2023-04-23"
+AND language = 'en'
+ORDER BY publishedAt
+LIMIT 40;
+```
+
+The query parameter in mandatory
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/newsapi_handler/README.md:59]: Can we please explain this better?

File: mindsdb/integrations/handlers/newsapi_handler/newsapi_handler.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                article["searchIn"] = params.get("searchIn")
+                article["domains"] = params.get("domains")
+                article["excludedDomains"] = params.get("exclude_domains")
+                data.append(article)
+
+        return pd.DataFrame(data=data)
+
+
+connection_args = OrderedDict(
+    api_key={"type": ARG_TYPE.STR, "description": "The API key for the Airtable API."}
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/newsapi_handler/newsapi_handler.py:215]: Please change the Airatable here

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite bot:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.<br/>
   - - -
   I have read the CLA Document and I hereby sign the CLA
   - - -
   <sub>You can retrigger this bot by commenting **recheck** in this Pull Request</sub>

File: mindsdb/integrations/handlers/newsapi_handler/newsapi_handler.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        self.is_connected = True
+        return HandlerStatusResponse(success=True)
+
+    def check_connection(self) -> HandlerStatusResponse:
+        response = HandlerStatusResponse(False)
+
+        try:
+            self.connect()
+
+            self.api.get_top_headlines()
```
[the remark this answers is not in the document]
Comment (KenmogneThimotee) [reply mindsdb/integrations/handlers/newsapi_handler/newsapi_handler.py:171]: This API has only 3 routes and no one is there to check the connection status or API  key validation. This is done when you fetch data.
I can reduce returned payload like this: `self.api.get_top_headlines(page_size=1, page=1)`

File: mindsdb/integrations/handlers/newsapi_handler/README.md (hunk 3 above)
Comment (KenmogneThimotee) [reply mindsdb/integrations/handlers/newsapi_handler/README.md:59]: No problem I will rewrite the readme to make it more clear.

**Comment by KenmogneThimotee** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**